web/modules/custom/joinup_community_content/src/Access/NodeRevisionAccessCheck.php

```diff
@@ -66,8 +66,18 @@ public function access(Route $route, AccountInterface $account, $node_revision =
       return $og_access;
     }
 
-    $global_access = AccessResult::allowedIf($node && $this->checkAccess($node, $account, $operation));
-    return $global_access->cachePerPermissions()->addCacheableDependency($node)->inheritCacheability($og_access);
+    if ($operation === 'view') {
+      $global_access = AccessResult::allowedIf($node && (
+          $this->checkAccess($node, $account, $operation)
+          || ($node->getOwnerId() === $account->id() && $account->hasPermission('view own content revisions'))
+        )
+      );
```

Comment:
I had a call with @idimopoulos about this and it is more clear to me now. This is only about granting `view` access, which is ignored by OG. The `edit` and `delete` access is fully handled by core.

This can probably be made fully robust by just checking if the user has view permission on the entity, and additionally has the `view own content revisions` permission.